DISCOURSE.md

```diff
@@ -0,0 +1,67 @@
+  - A webserver like nginx to act as reverse proxy so that the outside world can talk with loginbot web APIs.
+4. A domain you can point to the webserver
+
+## Steps
+
+The first step assumes you are on Linux and you are fine with a statically linked binary with musl libc.
+
+ 1. Download the latest release from Releases and get the binary: `wget https://github.com/deltachat-bot/deltachat-loginbot/releases/download/v0.3.0/deltachat-loginbot-x86_64-linux-musl_0.3.0.tar.bz2`
+ 2. Extract loginbot's binary and other stuff to an empty directory: `tar xvf deltachat-loginbot-x86_64-linux-musl_0.3.0.tar.bz2 -C empty_directory`
+ 3. Rename `example_config.toml` to `config.toml` and modify the TOML config file(`config.toml`) to your need. Enter the email address and password for the loginbot account into the `email` and `password` fields in `config.toml`
+ 4. Use `bash scripts/gen_secret.sh` to generate two random strings for `client_id` and `client_secret` in `oauth` section of the config file. Enter one of randomly generated strings in `client_id` field and the other in `client_secret` field.
+ 5. Enter `https://discourse.tld/auth/oauth2_basic/callback` as `redirect_uri` where `discouse.tld` is the domain address of your Discourse.
+ 6. Run the binary. By default, it looks for `config.toml` in the current directory but you can specify somewhere else by a command line argument: `./loginbot /path/to/config.toml`
+ 7. For production runs, a service manager is highly recommended for easier management of the loginbot process. For instance on Debian based distros and many others, systemd is used. You can use the systemd user service template(`loginbot.service`).
+ 8. The bot's web API is listening to `listen_addr` as specified in the config file. To see different possible values for `listen_addr` see [valid values for SocketAddr](https://doc.rust-lang.org/nightly/core/net/enum.SocketAddr.html).
+9. You need a webserver to act as a reverse proxy for the login bot (see [this guide](https://www.digitalocean.com/community/tutorials/how-to-configure-nginx-as-a-reverse-proxy-on-ubuntu-22-04) as an example). Let it proxy traffic to the `listen_addr` from the `config.toml`.
+10. Point your domain to the server, so users can reach it (e.g. with `foo.com`).
+11. Create a TLS certificate for your domain, so the reverse proxy can encrypt the connection to the user, e.g. with [certbot](https://certbot.eff.org/).
+9. In your Discourse instance, navigate to the admin panel. Then open Site settings and then "Login" section.
+10. Install [Discourse basic OAuth2 plugin](https://github.com/discourse/discourse-oauth2-basic) see plugin installation guide [here](https://github.com/deltachat-bot/deltachat-loginbot/pull/16/files).
+11. Configure Basic OAuth2:
+```
+oauth2 enabled: 			true
+oauth2 client id: 			secret
+oauth2 client secret: 			secret
+oauth2 authorize url:			https://login.testrun.org/authorize
+oauth2 token url:			https://login.testrun.org/token
```

Comment:
```suggestion
oauth2 client id: 			<secret>
oauth2 client secret: 			<secret>
oauth2 authorize url:			https://foo.com/authorize
oauth2 token url:			https://foo.com/token
```

We used `foo.com` consistently through the guide, `login.testrun.org` might confuse the reader.

Reply:
actually changing every occurence of `foo.com` to `login.example.com` might be even better. But it should be consistent at least.